.github/workflows/ci.yml

```diff
@@ -0,0 +1,145 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+
+
+jobs:
+  sanity:
+    name: ${{ matrix.test.name }}
+    runs-on: ubuntu-20.04
+    container:
+      image: quay.io/ansible/ansible-runner-test-container:2.0.0
+      env:
+        PIP_CACHE_DIR: ${{ runner.temp }}/.cache/pip
+        PY_COLORS: 1
+        TOXENV: ${{ matrix.test.tox_env }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        test:
+          - name: Lint
+            tox_env: linters
+
+          - name: Docs
+            tox_env: docs
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Create tox environment
+        run: tox --notest
+
+      - name: Run tests
+        run: tox
+
+
+  images:
+    name: Build Image - ${{ matrix.runtime }}
+    runs-on: ubuntu-20.04
+
+    strategy:
+      fail-fast: false
+      matrix:
+        runtime:
+          - docker
+          - podman
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Build Container Image
+        run: ${{ matrix.runtime }} build --rm=true -t quay.io/ansible/ansible-runner:devel -t quay.io/ansible/ansible-runner:latest .
+
+
+  integration:
+    runs-on: ubuntu-20.04
+    name: Integration - ${{ matrix.py_version.name }}
+    needs: images
```

Comment:
I have been frustrated by the lack of artifacts support with Github Actions myself. My understanding of "needs:" is that it is strictly temporal. As in, it will make a step wait until a prior check is finished, but it doesn't carry over anything from the file system of the prior job.

I've seen some plugins that implement custom means of passing artifacts, but they're ugly. These tend to push the artifacts to some datastore and then pull it again in the job that uses it. Yes, `needs:` would play a role there to get the timing right, but in this case you would still need to pull and pull from an image registry, if I understand correctly.

Reply:
That might explain why the GHA integration tests did not fail as I expected them too with this change. If this is the case, then I don't think switching to GHA for the tests is a viable thing to do.

Reply:
FYI I used GH container registry for artifacts in the creator-ee repo, you may want to use it as an example. The trick is to bake in the workflow run ID so that you could publish from one job and pick up from the next stage. See the release workflow specifically for this example.

Reply:
Or did you mean normal file artifacts? There's official upload-/download-artifacts actions for that.